homeassistant/components/transmission/sensor.py

```diff
@@ -28,23 +31,103 @@
-    SensorEntityDescription(key="paused_torrents", translation_key="paused_torrents"),
-    SensorEntityDescription(key="total_torrents", translation_key="total_torrents"),
-    SensorEntityDescription(
-        key="completed_torrents", translation_key="completed_torrents"
+MODES: dict[str, list[str] | None] = {
+    "started_torrents": ["downloading"],
+    "completed_torrents": ["seeding"],
+    "paused_torrents": ["stopped"],
+    "active_torrents": [
+        "seeding",
+        "downloading",
+    ],
+    "total_torrents": None,
+}
+
+
+@dataclass(kw_only=True)
+class TransmissionSensorEntityDescription(SensorEntityDescription):
+    """Entity description class for Transmission sensors."""
+
+    val_func: Callable[[TransmissionDataUpdateCoordinator], StateType]
+    extra_state_attr_func: Callable[[Any], dict[str, str]] | None = None
+
+
+SENSOR_TYPES: tuple[TransmissionSensorEntityDescription, ...] = (
+    TransmissionSensorEntityDescription(
+        key="download",
+        translation_key="download_speed",
+        device_class=SensorDeviceClass.DATA_RATE,
+        native_unit_of_measurement=UnitOfDataRate.BYTES_PER_SECOND,
+        suggested_display_precision=2,
+        suggested_unit_of_measurement=UnitOfDataRate.MEGABYTES_PER_SECOND,
+        val_func=lambda coordinator: float(coordinator.data.download_speed),
+    ),
+    TransmissionSensorEntityDescription(
+        key="upload",
+        translation_key="upload_speed",
+        device_class=SensorDeviceClass.DATA_RATE,
+        native_unit_of_measurement=UnitOfDataRate.BYTES_PER_SECOND,
+        suggested_display_precision=2,
+        suggested_unit_of_measurement=UnitOfDataRate.MEGABYTES_PER_SECOND,
+        val_func=lambda coordinator: float(coordinator.data.upload_speed),
+    ),
+    TransmissionSensorEntityDescription(
+        key="status",
+        translation_key="transmission_status",
+        device_class=SensorDeviceClass.ENUM,
+        options=[STATE_IDLE, STATE_UP_DOWN, STATE_SEEDING, STATE_DOWNLOADING],
+        val_func=lambda coordinator: get_state(
+            coordinator.data.upload_speed, coordinator.data.download_speed
+        ),
+    ),
+    TransmissionSensorEntityDescription(
+        key="active_torrents",
+        translation_key="active_torrents",
+        native_unit_of_measurement="Torrents",
+        val_func=lambda coordinator: coordinator.data.active_torrent_count,
+        extra_state_attr_func=lambda coordinator: _torrents_info_attr(
+            coordinator=coordinator, key="active_torrents"
+        ),
+    ),
+    TransmissionSensorEntityDescription(
+        key="paused_torrents",
+        translation_key="paused_torrents",
+        native_unit_of_measurement="Torrents",
+        val_func=lambda coordinator: coordinator.data.paused_torrent_count,
+        extra_state_attr_func=lambda coordinator: _torrents_info_attr(
+            coordinator=coordinator, key="paused_torrents"
+        ),
```

Comment:
I think it would be good to move these torrent lists to a service call now we support service responses. You can have quite some torrents, and it will kinda blow up your state machine.

For a follow up PR btw